tools/cli-generator/generate.go

```diff
@@ -0,0 +1,281 @@
+	Method     string `yaml:"method,omitempty"`
+	SDKMethod  string `yaml:"sdk_method,omitempty"`
+	ArgName    string `yaml:"arg_name,omitempty"`
+	ArgType    string `yaml:"arg_type,omitempty"`
+	ReturnType string `yaml:"return_type,omitempty"`
+}
+
+func (c *Command) CommandPaths() []string {
+	return strings.Split(c.CommandPath, " ")
+}
+
+func (c *Command) LastCommandPath() string {
+	commandPaths := c.CommandPaths()
+
+	return commandPaths[len(commandPaths)-1]
+}
+
+func (c *Command) baseFileName(basePath string) string {
+	if len(c.SubCommands) > 0 {
+		return filepath.Join(basePath, "internal", "cli", filepath.Join(c.CommandPaths()...), c.LastCommandPath())
+	}
+
+	return filepath.Join(basePath, "internal", "cli", filepath.Join(c.CommandPaths()...))
+}
+
+func (c *Command) FileName(basePath string) string {
+	return c.baseFileName(basePath) + ".go"
+}
+
+func (c *Command) UnitTestFileName(basePath string) string {
+	return c.baseFileName(basePath) + "_test.go"
+}
+
+func fileExists(f string) bool {
+	_, err := os.Stat(f)
+	return !os.IsNotExist(err)
+}
+
+func (cli *CLI) generateStore(store *Store) error {
+	storeFile := filepath.Join(cli.basePath, "internal", "store", store.BaseFileName+".go")
+
+	fileCreated, err := cli.generateFile(storeFile, store.TemplateFile(), store)
+	if err != nil {
+		return err
+	}
+	if !fileCreated {
+		return nil
+	}
+	return goGenerate(storeFile)
+}
+
+func (cli *CLI) generateFile(file string, templateFile string, data any) (bool, error) {
+	if !cli.overwrite && fileExists(file) {
+		fmt.Printf("File '%s' already present in disk, skipping\n", file)
+		return false, nil
+	}
+
+	err := os.MkdirAll(filepath.Dir(file), os.ModePerm)
+	if err != nil {
+		return false, err
+	}
+
+	f, err := os.Create(file)
+	if err != nil {
+		return false, err
+	}
+
+	defer f.Close()
+
+	err = cli.templates.ExecuteTemplate(f, templateFile, data)
+	if err != nil {
+		return false, err
+	}
+
+	err = cleanupFile(file)
+	if err != nil {
+		return true, err
+	}
+
+	return true, nil
+}
+
+func (cli *CLI) generateCommand(cmd *Command) error {
+	_, err := cli.generateFile(cmd.FileName(cli.basePath), cmd.TemplateFile(), cmd)
+	if err != nil {
+		return err
+	}
+
+	_, err = cli.generateFile(cmd.UnitTestFileName(cli.basePath), cmd.TemplateUnitTestFile(), cmd)
+	if err != nil {
+		return err
+	}
+
+	for i := range cmd.SubCommands {
+		if err := cli.generateCommand(&cmd.SubCommands[i]); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func cleanupFile(filePath string) error {
+	execCmd := exec.Command("goimports", "-w", filePath)
+	_, err := execCmd.Output()
+	if err != nil {
+		return err
+	}
+
+	execCmd = exec.Command("gofmt", "-w", "-s", filePath)
+	_, err = execCmd.Output()
+	return err
+}
+
+func goGenerate(filePath string) error {
+	execCmd := exec.Command("go", "generate", filePath)
+	_, err := execCmd.Output()
+	return err
+}
+
+func runMake() error {
+	execCmd := exec.Command("make", "gen-docs")
```

Comment:
👀 Copypaste? Thought it would be `gen-code`

Reply:
`gen-code` calls this process

Reply:
Ahhh got it, this is part of the call-chain